DR2-1257 Add update password methods

MancunianSam opened the pull request:
DR2-1257 Add methods to reset and test password changes
    
I've added a method in to reset the password and one to check we can get a token with the new password.

I've also replaced the direct calls to AWS secrets manager SDK with the da-aws-clients one. This meant I had to change all the implicit Sync for Async which is why there's more changes.

techncl reviewed (commented):
Looks good! Just one question, as well as:

1. When we were on the call with Preservica, they showed that you are able to give the old password as the new password for some reason (maybe the didn't think about it). Should we check for and prevent this in this just in case?
2. Should we add any other checks like length is not 0?

MancunianSam commented:
I've added in a check as well for 0 length on the new password and for when the old and new passwords are the same. I've added in a test to check for those things.

techncl reviewed (approved):
👍
